### PR TITLE
Doc update for latin1 decoder & encoder

### PR DESCRIPTION
**Pull request by jpsiitonen**
Adding example codes for classes: Latin1Encoder and Latin1Decoder
Class description update for Latin1Decoder

### CODE REVIEW

File: sdk/lib/convert/latin1.dart [hunk 1]
```diff
@@ -63,12 +63,54 @@ class Latin1Codec extends Encoding {
 }
 
 /// This class converts strings of only ISO Latin-1 characters to bytes.
+///
+/// Example:
+/// ```dart
+/// const String sample = 'à á â ã ä å';
```
Comment (alexlindroos) [inline sdk/lib/convert/latin1.dart]: Maybe we should remove the whitespace from sample so they are not converted to "32" and it's easier to spot 6 characters to convert to 6 different bytes

File: sdk/lib/convert/latin1.dart [hunk 2]
```diff
@@ -63,12 +63,54 @@ class Latin1Codec extends Encoding {
...
   const Latin1Encoder() : super(_latin1Mask);
 }
 
 /// This class converts Latin-1 bytes (lists of unsigned 8-bit integers)
 /// to a string.
+///
+/// Example:
+/// ```dart
+/// const Latin1Decoder latin1Decoder = Latin1Decoder();
+/// // dec values as source
```
Comment (alexlindroos) [inline sdk/lib/convert/latin1.dart]: what is "dec"? If thats abbreviation then we should use the longer word

File: sdk/lib/convert/latin1.dart [hunk 3]
```diff
@@ -63,12 +63,54 @@ class Latin1Codec extends Encoding {
...
+/// const List<int> latin1bytes = [
+///   224, 32, 225, 32, 226, 32, 227, 32, 228, 32, 229];
```
Comment (alexlindroos) [inline sdk/lib/convert/latin1.dart]: Same whitespace thing, should we exclude them?

File: sdk/lib/convert/latin1.dart (hunk 2 above)
Comment (alexlindroos) [reply sdk/lib/convert/latin1.dart]: capital letter

File: sdk/lib/convert/latin1.dart [hunk 4]
```diff
@@ -63,12 +63,54 @@ class Latin1Codec extends Encoding {
...
+/// final String string = latin1Decoder.convert(latin1bytes);
+/// print(string); // à á â ã ä å
+///
+/// // hex values as source
```
Comment (alexlindroos) [inline sdk/lib/convert/latin1.dart]: capital letter

File: sdk/lib/convert/latin1.dart [hunk 5]
```diff
@@ -63,12 +63,54 @@ class Latin1Codec extends Encoding {
...
+/// const List<int> hexBytes = [
+///   0xe0, 0x20, 0xe1, 0x20, 0xe2, 0x20, 0xe3, 0x20, 0xe4, 0x20, 0xe5];
```
Comment (alexlindroos) [inline sdk/lib/convert/latin1.dart]: whitespace or not?

File: sdk/lib/convert/latin1.dart [hunk 6]
```diff
@@ -63,12 +63,54 @@ class Latin1Codec extends Encoding {
...
+/// final decoded2 = latin1Decoder.convert(hexBytes);
```
Comment (alexlindroos) [inline sdk/lib/convert/latin1.dart]: can we think of a better variable name than "decoded2"?

File: sdk/lib/convert/latin1.dart [hunk 7]
```diff
@@ -63,12 +63,54 @@ class Latin1Codec extends Encoding {
...
+/// throws [FormatException].
+///
+///
+/// If `allowInvalid` is set to true on constructor,
+/// the decoder replaces the invalid bytes using a character `U+FFFD` �.
+///
+/// Example:
+/// ```dart
+/// const Latin1Decoder latin1Decoder = Latin1Decoder(allowInvalid: true);
+/// const List<int> encoded = [224, 32, 225, 32, 226, 32, 227, 32, 228, 32,
```
Comment (alexlindroos) [inline sdk/lib/convert/latin1.dart]: whitespace or not?

File: sdk/lib/convert/latin1.dart (hunk 1 above)
Comment (jpsiitonen) [reply sdk/lib/convert/latin1.dart]: Good point - will make that update 👍

File: sdk/lib/convert/latin1.dart [hunk 8]
```diff
@@ -63,12 +63,54 @@ class Latin1Codec extends Encoding {
...
 }
 
 /// This class converts Latin-1 bytes (lists of unsigned 8-bit integers)
 /// to a string.
+///
+/// Example:
+/// ```dart
+/// const Latin1Decoder latin1Decoder = Latin1Decoder();
+/// // dec values as source
+/// const List<int> latin1bytes = [
```
Comment (alexlindroos) [inline sdk/lib/convert/latin1.dart]: nit: if we have latin1Decoder, should we also have latin1Bytes instead

File: sdk/lib/convert/latin1.dart (hunk 2 above)
Comment (jpsiitonen) [reply sdk/lib/convert/latin1.dart]: yeah. basically not needed comment in code

**Comment by alexlindroos** [comment]
LGTM

File: sdk/lib/convert/latin1.dart [hunk 9]
```diff
@@ -63,12 +63,50 @@ class Latin1Codec extends Encoding {
...
+/// print(decoded); // àáâãäå
+///
+/// // Hexadecimal values as source
+/// const List<int> hexBytes = [0xe0, 0xe1, 0xe2, 0xe3, 0xe4, 0xe5];
+/// final String decodedHexBytes = latin1Decoder.convert(hexBytes);
+/// print(decodedHexBytes); // àáâãäå
+/// ```
+///
+/// If `bytes` contains values that are not in the range 0 .. 255, the decoder
+/// throws [FormatException].
```
Comment (tomgilder) [inline sdk/lib/convert/latin1.dart]: nit: should probably have an "a"

```suggestion
/// throws a [FormatException].
```

File: sdk/lib/convert/latin1.dart [hunk 10]
```diff
@@ -63,12 +63,50 @@ class Latin1Codec extends Encoding {
...
+///
+///
+/// If `allowInvalid` is set to true on constructor,
```
Comment (tomgilder) [inline sdk/lib/convert/latin1.dart]: This doesn't need "on constructor"

```suggestion
/// If `allowInvalid` is set to true,
```

File: sdk/lib/convert/latin1.dart [hunk 11]
```diff
@@ -63,12 +63,50 @@ class Latin1Codec extends Encoding {
...
+/// the decoder replaces the invalid bytes using a character `U+FFFD` �.
```
Comment (tomgilder) [inline sdk/lib/convert/latin1.dart]:
```suggestion
/// the decoder replaces the invalid bytes with the character `U+FFFD` �.
```

File: sdk/lib/convert/latin1.dart [hunk 12]
```diff
@@ -63,12 +63,50 @@ class Latin1Codec extends Encoding {
...
+///
+///
+/// Allow invalid byte value example:
```
Comment (tomgilder) [inline sdk/lib/convert/latin1.dart]:
```suggestion
/// Example with `allowInvalid` set to true:
```

**Review by tomgilder — APPROVED** [review]

**Review by alexlindroos — APPROVED** [review]

**Review by stepthomp — APPROVED** [review]

**Pull request closed without merging**